storage/indexer/src/indexer_reader.rs

```diff
@@ -44,9 +44,11 @@ impl IndexerReader for IndexerReaders {
         self.db_indexer_reader.is_some()
     }
 
+    /// A blocking call that waits for the internal indexer to catch up to the given version
```

Comment:
One thing we might want to do now or later is: we probably want to differentiate the table handle is not found in db because of the internal indexer is falling behind v.s. the table handle is not found because of it is invalid..